lib/ide/src/controller/module.rs

```diff
@@ -0,0 +1,64 @@
+//! Module Controller.
+//!
+//! The module controller keeps cached module state (module state is AST+Metadata or equivalent),
+//! and uses it for synchronizing state for text and graph representations. It provides method
+//! for registering text and graph changes. If for example text represntation will be changed, there
+//! will be notifications for both text change and graph change.
+//!
+//! This module is still on WIP state, for now it contains stubs only.
+
+use crate::prelude::*;
+
+use shapely::shared;
+
+
+/// Structure uniquely identifying module location in the project.
+/// Mappable to filesystem path.
+#[derive(Clone,Debug,Eq,Hash,PartialEq)]
+pub struct Location(pub String);
+
+impl Location {
+    /// Obtains path (within a project context) to the file with this module.
+    pub fn to_path(&self) -> file_manager_client::Path {
+        // TODO [mwu] Extremely provisional. When multiple files support is
+        //            added, needs to be fixed, if not earlier.
+        let Location(string) = self;
+        let result = format!("./{}.luna", string);
```

Comment:
Please do not encode product name here. Extension of source files should be refactored to some top-level module, which is easy to find and cointains all literals for configuration of special names. Moreover, Marcin is already using `.enso` extension.